### DISCUSSION TITLE
Please expose `sassLoaderOptions` rather than or in addition to `sassOptions`

### DISCUSSION THREAD

**Discussion by lunelson** [vercel/next.js#24105] [Ideas]
### Describe the feature you'd like to request

The [`sassOptions` property in `next.config.js`](https://nextjs.org/docs/basic-features/built-in-css-support#customizing-sass-options) is not really enough for advanced Sass configuration with Webpack, as the only way to reliably set up global variables, mixins etc. across multiple entry points is to inject a custom header using [the `additionalData` option exposed by **sass-loader**](https://github.com/webpack-contrib/sass-loader#additionaldata), currently unreachable.

### Describe the solution you'd like

Please expose the `sassLoaderOptions` rather than just `sassOptions`. The former contains the latter, and `sassOptions` could also remain, so the following would be equivalent:

```js

// assuming `mySassOptions` exists, the following are equivalent:

// old form
module.exports = {
  sassOptions: mySassOptions,
}

// new form
module.exports = {
  sassLoaderOptions: {
    sassOptions: mySassOptions,
  }
}
```

### Describe alternatives you've considered

Currently I use a custom plugin to do this, but it is brittle:

```js
// assuming I have already created `myCustomSassLoaderOptions`

module.exports = ({ configPath, isDev } = {}) => (nextConfig = {}) => {
  return Object.assign({}, nextConfig, {
    webpack(config, options) {
      config.module.rules.forEach((moduleRule) => {
        if (!moduleRule.oneOf) return;
        moduleRule.oneOf.forEach((loaderRule) => {
          if (!Array.isArray(loaderRule.use)) return;
          loaderRule.use.forEach((useRule) => {
            if (!useRule.loader.match('sass-loader')) return;
            useRule.options = myCustomSassLoaderOptions;
          });
        });
      });

      if (typeof nextConfig.webpack === 'function') {
        return nextConfig.webpack(config, options);
      }

      return config;
    },
  });
};
```

**Comment by timneutkens** [comment]
You can add `additionalData` to `sassOptions` already, so this is not needed.

**Comment by lunelson** [comment]
@timneutkens now that you say that, after some digging I found it:

https://github.com/vercel/next.js/blob/1c2ee7c6b8de144ffcdbbde68abf73570839dbd5/packages/next/build/webpack/config/blocks/css/index.ts#L30-L36

However, since this is totally undocumented, and non-standard WRT `sass` and `sass-loader` APIs, I think it deserves more of a response than a summary closure. 

The documentation says that `sassOptions` can be used to "configure the Sass compiler". `additionalData` is not a `sass` option, it's a `sass-loader` option. How would anyone be expected to know this is available?
